File: solutions/python/line-up/1/line_up.py
def line_up(name, number):
    last_digit = number % 10
    last_two = number % 100
    number_str = str(number)
    if last_two in {11,12,13}:
        number_str += "th"
    else:
        match last_digit:
            case 1:
                number_str += "st"
            case 2:
                number_str += "nd"
            case 3:
                number_str += "rd"
            case _:
                number_str += "th"
    return f"{name}, you are the {number_str} customer we serve today. Thank you!"

File: solutions/free-pascal/line-up/1/LineUp.pas
unit LineUp;

{$mode ObjFPC}{$H+}

interface

function FormatMessage(
  const name : ShortString; const number : byte
) : ShortString;

implementation

uses SysUtils;

function FormatMessage(
  const name : ShortString; const number : byte
) : ShortString;
var 
  lastDigit, lastTwo : byte;
  numberStr : string;
begin
  lastDigit := number mod 10;
  lastTwo := number mod 100;
  numberstr:= IntToStr(number);
  if (lastTwo = 11) or (lastTwo = 12) or (lastTwo = 13) then
  begin
    numberStr += 'th';
  end
  else
  begin
    case lastDigit of 
      1 : numberStr += 'st';
      2 : numberStr += 'nd';
      3 : numberStr += 'rd';
      0,4..9 : numberStr += 'th';
    end;
  end;  
  result := name + ', you are the ' + numberStr + ' customer we serve today. Thank you!'
end;

end.
